packages/build-tools/src/utils/expoUpdates.ts

```diff
@@ -88,14 +94,36 @@ export async function configureExpoUpdatesIfInstalledAsync(
   }
 
   const appConfigRuntimeVersion = ctx.job.version?.runtimeVersion ?? resolvedRuntimeVersion;
+
   if (ctx.metadata?.runtimeVersion && ctx.metadata.runtimeVersion !== appConfigRuntimeVersion) {
     ctx.markBuildPhaseHasWarnings();
-    ctx.logger.warn(
-      `Runtime version from the app config evaluated on your local machine (${ctx.metadata.runtimeVersion}) does not match the one resolved here (${appConfigRuntimeVersion}).`
-    );
-    ctx.logger.warn(
-      "If you're using conditional app configs, e.g. depending on an environment variable, make sure to set the variable in eas.json or configure it with EAS Secret."
-    );
+    ctx.logger.warn('Runtime version mismatch');
+    ctx.logger.warn(`Runtime version on your local machine: ${ctx.metadata.runtimeVersion}`);
+    ctx.logger.warn(`Runtime version calculated on EAS: ${appConfigRuntimeVersion}`);
+
+    if (ctx.metadata?.fingerprintSource && resolvedFingerprintSources && resolvedRuntimeVersion) {
+      try {
+        const fingerprintSource = ctx.metadata?.fingerprintSource;
+
+        if (fingerprintSource.type === FingerprintSourceType.URL) {
+          const result = await fetch(fingerprintSource.url);
+          const localFingerprint = await result.json();
+          const easFingerprint = {
+            hash: resolvedRuntimeVersion,
+            sources: resolvedFingerprintSources as FingerprintSource[],
```

Comment:
Might be cleaner to hoist this case all the way up to `resolveRuntimeVersionAsync` and use `FingerprintSource[]` instead of `object[]` as types everywhere. 

Optionally could also add validation in `resolveRuntimeVersionAsync` to check that `runtimeVersionResult.fingerprintSources` adheres to the expected type before returning it using a library like `zod`. This would be a good stacked PR though as it's a bit outside the scope of this PR.